### PR TITLE
fix several gregor recent gregor regressions in stateByCategoryPrefix

### PR DESCRIPTION
**Pull request by maxtaco**
- burned by the nil interface v nil type bug on deviceID and timeOrOffset, giving us nonsensical query parameters
- another regression -- since we are threading clockwork's real clock through more code, we were usually not using NOW() on mysql, which gave us inaccurate time stamps.  I'm not sure when that bug snuck in, but i came across it as i was debugging queries.

### CODE REVIEW

File: rpc/server/main.go [hunk 1]
```diff
@@ -601,11 +601,18 @@ func (s *Server) storageDispatchHandler() {
 		case stateReq:
 			var res gregor.State
 			var err error
+			var deviceID gregor.DeviceID
+			if req.deviceID != nil && len(req.deviceID) > 0{
+				deviceID = req.deviceID
```
Comment (maxtaco) [inline rpc/server/main.go]: You hit the *worst* feature of Go.  Welcome to the club!

**Comment by mmaxim** [comment]
LGTM!

**Comment by maxtaco** [comment]
Btw here is the bug in play: https://golang.org/doc/faq#nil_error

**Comment by maxtaco** [comment]
We really need a test for this.  I'll add it in a separate PR.

**Pull request merged**